ttnn/cpp/ttnn/operations/matmul/matmul.cpp

```diff
@@ -44,11 +37,12 @@ std::optional<UnaryWithParam> get_fused_activation(const std::optional<const std
     return ttnn::operations::unary::utils::string_to_unary_with_param(activation.value());
 }
 
-ttnn::Tensor matmul(
+ttnn::Tensor bound_matmul(
```

Comment:
why is it called bound_matmul now?

Reply:
Because now the matmul in matmul_op.hpp/cpp is in the same namespace.

If I don't rename this method then I get compile time errors:
```
../ttnn/cpp/ttnn/operations/matmul/matmul_pybind.hpp:166:20: error: call to 'matmul' is ambiguous
  166 |             return matmul(
      |                    ^~~~~~
../ttnn/cpp/ttnn/operations/matmul/device/matmul_op.hpp:183:8: note: candidate function
  183 | Tensor matmul(
      |        ^
../ttnn/cpp/ttnn/operations/matmul/matmul.hpp:31:14: note: candidate function
   31 | ttnn::Tensor matmul(
      |       
```

I couldn't think of a better name than bound_matmul, to indicate that this method is called directly from pybind. I'm open to suggestions.